sdk/servicebus/service-bus/samples/typescript/src/receiveMessagesStreaming.ts

```diff
@@ -30,26 +36,49 @@ export async function main() {
   // - See session.ts for how to receive using sessions.
   const receiver = sbClient.createReceiver(queueName);
 
-  const processMessage = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-  const processError = async (args) => {
-    console.log(`Error from error source ${args.errorSource} occurred: `, args.error);
-  };
-
   try {
-    receiver.subscribe(
+    const subscription = receiver.subscribe(
       {
-        processMessage,
-        processError
+        processMessage: async (brokeredMessage: ServiceBusReceivedMessage) => {
+          console.log(`Received message: ${brokeredMessage.body}`);
+          await receiver.completeMessage(brokeredMessage);
+        },
+        processError: async (args: ProcessErrorArgs) => {
+          console.log(`Error from source ${args.errorSource} occurred: `, args.error);
+
+          // the handler will not stop without explicit intervention from you.
+          if (isMessagingError(args.error)) {
+            switch (args.error.code) {
+              case "MessagingEntityDisabledError":
+              case "MessagingEntityNotFoundError":
+              case "UnauthorizedError":
+                // It's possible you have a temporary infrastructure change (for instance, the entity being
+                // temporarily disabled). The handler will continue to retry - it is completely up to you
+                // what is considered fatal for your program.
+                console.log(
+                  `An unrecoverable error occurred. Stopping processing. ${args.error.code}`,
+                  args.error
+                );
+                await subscription.close();
+                break;
+              case "MessageLockLostError":
+                console.log(`Message lock lost for message`, args.error);
+                break;
+              case "ServerBusyError":
+                // choosing an arbitrary amount of time to wait.
+                await delay(1000);
+                break;
+            }
+          }
+        }
       },
       {
         autoComplete: false
```

Comment:
Hmm.. I no longer believe that setting autoComplete to false is the thing we want to advertise.
If the idea was that we want to show how to complete the message, then we do that in the other sample where we receive in batches or use the iterator right?

Reply:
I agree - our emphasis on subscribe() being the absolute simplest path

Reply:
Let's keep it so that the users would know the bare minimum thing to do in the precessMessage handler.. i.e., completing the message.

Reply:
> I think, however, I'll still show the setting and just set it to 'true' just to point out that it exists and is happening.

We can do that using comments in the sample code as well

```js
processMessage: async (brokeredMessage: ServiceBusReceivedMessage) => {
        console.log(`Received message: ${brokeredMessage.body}`);
        // Your code to process the message goes here after which the client will automatically complete the message.
        // If you want to hold on to the message and settle it later, set `autoComplete` to `false` in the options passed to `subscribe()`
}
```

Reply:
omments added.